float8_experimental/float8_tensor_parallel.py

```diff
@@ -0,0 +1,103 @@
+import torch.nn as nn
+from torch.distributed._tensor import DTensor
+from torch.distributed.device_mesh import DeviceMesh
+from torch.distributed.tensor.parallel import ColwiseParallel, RowwiseParallel
+
+# subclass the ColwiseParallel and RowwiseParallel classes
+# to add the float8 support
+# The parameter sharding stays the same as the core
+# ColwiseParallel and RowwiseParallel, the only difference
+# here is that in input/output handling we do casting after
+# creating the DTensor.
+
+# NOTE: This only works and tested with the DynamicLinear
```

Comment:
Out of curiosity do we have a path for Delayed?

Reply:
I'm not quite familiar with delayed scaling yet, but from a quick look I think the approach we are building with dynamic scaling might be working with delayed scaling, might need some more changes and testing :)